enclave/src/main.rs

```diff
@@ -0,0 +1,160 @@
+use byteorder::{NativeEndian, WriteBytesExt};
+use chrono::prelude::*;
+use std::io::prelude::*;
+use std::{fs, path};
+
+mod bpf;
+use bpf::*;
+
+const RUNC_CHILD: &str = "runc:[2:INIT]";
+
+#[derive(thiserror::Error, Debug)]
+enum CheckBpfLsmError {
+    #[error("regex compilation error")]
+    RegexError(#[from] regex::Error),
+    
+    #[error("I/O error")]
+    IOError(#[from] std::io::Error),
+
+    #[error("BPF LSM is not enabled")]
+    BpfLsmDisabledError,
+}
+
+fn check_bpf_lsm_enabled() -> Result<(), CheckBpfLsmError> {
+    let rx = regex::Regex::new(r"bpf")?;
+    let path = path::Path::new("/sys").join("kernel").join("security").join("lsm");
+    let mut file = fs::File::open(path)?;
+    let mut content = String::new();
+    
+    file.read_to_string(&mut content)?;
+
+    if !rx.is_match(&content) {
+	return Err(CheckBpfLsmError::BpfLsmDisabledError);
+    }
+
+    Ok(())
```

Comment:
You could probably match this instead.
```rust
match rx.is_match(&content) {
    true => Ok(()),
    false => Err(CheckBpfLsmError::BpfLsmDisabledError),
}
```

Reply:
Is this more idiomatic and/or do we get some clarity or optimization from this?

Reply:
Idiomatic: preferring patterns to statements when possible to reduce verbosity and increase readability.

Since we know that the previous statement satisfies the returning type (`Result`), then we can reduce 1 statement and 1 pattern to one 1 pattern in total via a `match` statement.

Reply:
Solid. Thanks.